File: sdk/src/main/java/com/robotemi/sdk/listeners/OnWelcomingModeStatusChangedListener.java
package com.robotemi.sdk.listeners;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public interface OnWelcomingModeStatusChangedListener {

    String ACTIVE = "active";

    String IDLE = "idle";

    /**
     * Listen for status changes during 'Welcoming mode'.
     * <p>
     * Available status:
     * <ul>
     * <li>{@link OnWelcomingModeStatusChangedListener#ACTIVE}</li>
     * <li>{@link OnWelcomingModeStatusChangedListener#IDLE}</li>
     * </ul>
     *
     * @param status Current status.
     */
    void onWelcomingModeStatusChanged(@WelcomingModeStatus String status);

    @StringDef({ACTIVE, IDLE})
    @Retention(RetentionPolicy.SOURCE)
    @interface WelcomingModeStatus {
    }
}

File: sdk/src/main/java/com/robotemi/sdk/listeners/OnWelcomingModeStatusChangedListener.kt
package com.robotemi.sdk.listeners

import androidx.annotation.StringDef

interface OnWelcomingModeStatusChangedListener {

    /**
     * Listen for status changes during 'Welcoming mode'.
     *
     *
     * Available status:
     *
     *  * [OnWelcomingModeStatusChangedListener.ACTIVE]
     *  * [OnWelcomingModeStatusChangedListener.IDLE]
     *
     *
     * @param status Current status.
     */
    fun onWelcomingModeStatusChanged(@WelcomingModeStatus status: String)

    @StringDef(ACTIVE, IDLE)
    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    annotation class WelcomingModeStatus

    companion object {

        const val ACTIVE = "active"

        const val IDLE = "idle"
    }
}
